applications/FluidDynamicsApplication/custom_strategies/strategies/residualbased_predictorcorrector_velocity_bossak_scheme_turbulent.h

```diff
@@ -328,8 +336,21 @@ namespace Kratos {
 
                         array_1d<double, 3 > & OldVelocity = (itNode)->FastGetSolutionStepValue(VELOCITY, 1);
 
-                        noalias(itNode->FastGetSolutionStepValue(MESH_VELOCITY)) = itNode->FastGetSolutionStepValue(VELOCITY);
-                        UpdateDisplacement(CurrentDisplacement, OldDisplacement, OldVelocity, OldAcceleration, CurrentAcceleration);
+    
+                        
+                        if((itNode)->FastGetSolutionStepValue(IS_LAGRANGIAN_INLET) == 0.0)
```

Comment:
I'd suggest using a flag instead of a nodal variable, it will save you memory.

Reply:
Greetings dear @jcotela 

I hope I have done the required changes so far correctly, where I removed the (.directory), (.bin), and changed the header,

noting also please that I tried to follow your valuable recommendations for merging the master into this branch to fruther resolve all the conflicts, and I think it worked fine,, 

regarding using the flag instead the nodal variables, I agree with you, but kindly I will just work it out with Dr. Jarauta since he is the one who developed this code, and will include it soon.

Thanks and best regards,

Reply:
Greetings dear @jcotela and very good day to you,

Kindly, we need your feedback regarding the flag: 

1. could you please elaborate more about the recommended way to be used within kratos, because from what I read in kratos forum website "http://kratos-wiki.cimne.upc.edu/forum/index.php?topic=36.0" that it is not recommended to add/define a local flag variable, and also I was thinking that it will not be convenient to link it with fluid dynamic applications.

2. For example, can it be defined in similar manner:
    ```
    bool is_inlet = false;
    is_inlet = (itNode) -> FastGetSolutionStepValue(IS_LAGRANGIAN_INLET) == 0;
    if(is_inlet ==  true)
```


Thanks and our best regards,

Reply:
@elafnm: ideally, you would use one of the existing global flags (see post 4 in your link) instead of a nodal variable. If you don't think they are convenient for you, then just keep the current implementation. I also agree that adding a new local flag is not a good solution in this case.

Reply:
Greetings dear @jcotela and very good day to you,

If you do not mind please, let us just keep the current implementation, this would be more convenient to us.

thank you for your usual support and very positive feedback,